File: python/lesson 8/bfs_graph.py
class BfsGraph:
    def traverse(self, graph: Graph):
        visited = [False] * (graph.size())
        for v in range(0, graph.size() - 1):
            if not visited[v]:
                self.bfs(graph, v, visited)
            
    def bfs(self, graph: Graph, start: int, visited: List[bool]):
        queue = deque()
        
        visited[start] = True
        queue.append(start)
        print(start)
        
        while len(queue) > 0:
            v = queue.popleft()
            for w in graph.get(v):
                if not visited[w]:
                    visited[w] = True
                    queue.append(w)
                    print(w)

File: csharp/lesson 8/BfsGraph.cs
public class BfsGraph
{
    public void Bfs(Graph graph)
    {
        var visited = new bool[graph.Size];
        for (var v = 0; v < graph.Size; v++)
        {
            if (!visited[v]) Bfs(graph, v, visited);
        }
    }

    private void Bfs(Graph graph, int start, bool[] visited)
    {
        var queue =  new Queue<int>();
        
        visited[start] = true;
        queue.Enqueue(start);
        Console.WriteLine(start);

        while (queue.Count > 0)
        {
            var v = queue.Dequeue();
            foreach (var w in graph.Get(v))
            {
                if (!visited[w])
                {
                    visited[w] = true;
                    queue.Enqueue(w);
                    Console.WriteLine(w);
                }
            }
        }
    }
}
